code/modules/atmospherics/portable/portable_pressurizer.dm

```diff
@@ -0,0 +1,429 @@
+			src.blast_armed = FALSE
+		else
+			var/blast_key = rand()
+			blast_armed = blast_key
+			SPAWN_DBG(blast_delay)
+				if(src.blast_armed == blast_key && src.is_air_safe())
+					blast_release()
+
+	proc/blast_visual_effects(pressure)
+		// Perform jump animation if more than 70% maximum pressure
+		if(pressure > maximum_pressure * BLAST_EFFECT_RATIO)
+			var/orig_x = src.pixel_x
+			var/orig_y = src.pixel_y
+			animate(src, pixel_x=orig_x, pixel_y=orig_y, flags=ANIMATION_PARALLEL, time=0.01 SECONDS)
+			for(var/i in 1 to 3)
+				animate(pixel_x=orig_x + rand(-2, 2), pixel_y=orig_y + rand(-2, 2), easing=JUMP_EASING, time=0.1 SECONDS)
+			animate(pixel_x=orig_x, pixel_y=orig_y)
+
+		var/obj/overlay/poof = new/obj/overlay(get_turf(src))
+		poof.icon = 'icons/obj/atmospherics/atmos.dmi'
+		poof.color=rgb(air_contents.toxins/TOTAL_MOLES(air_contents)*255, 	\
+						air_contents.oxygen/TOTAL_MOLES(air_contents)*255,	\
+						air_contents.oxygen+air_contents.toxins/TOTAL_MOLES(air_contents)*255)
+		poof.alpha = clamp(MIXTURE_PRESSURE(src.air_contents)/src.maximum_pressure*180, 90, 220)
+		flick("pressurizer-poof", poof)
+		SPAWN_DBG(0.8 SECONDS)
+			if(poof) qdel(poof)
+
+	proc/blast_release()
+		var/pressure = MIXTURE_PRESSURE(src.air_contents)
+		src.blast_visual_effects(pressure)
+
+		// Flashbang pressure wave is 30,000 psi thus 206 MPa
+		var/volume = clamp(pressure KILO PASCALS / 206 MEGA PASCAL * 35, 15, 70)
+		playsound(src, "sound/effects/exlow.ogg", volume, 1)
+
+		var/turf/simulated/T = get_turf(src)
+		if(T && istype(T))
+			if(T.air)
+				if(T.parent?.group_processing)
+					T.parent.air.merge(src.air_contents)
+				else
+					var/count = length(T.parent?.members)
+					if(count)
+						if(count>1)
+							src.air_contents = src.air_contents.remove_ratio(count-1/count)
+						var/datum/gas_mixture/GM
+						for(var/turf/simulated/MT as() in T.parent.members)
+							GM = unpool(/datum/gas_mixture)
+							GM.copy_from(src.air_contents)
+							MT.assume_air(GM)
+					else
+						T.assume_air(src.air_contents)
+
+			if(pressure > (maximum_pressure * BLAST_EFFECT_RATIO))
+				for(var/mob/living/HH in range(8, src))
+					var/checkdist = get_dist(HH.loc, T)
+
+					// Reduced sonic boom effect with increased misstep from shockwave
+					var/misstep = clamp(1 + 10 * (5 - checkdist), 0, 40)
+					var/ear_damage = max(0, (3 - checkdist))
+					var/ear_tempdeaf = max(0, (5 - checkdist))
+					var/stamina = clamp(30 * (7 - checkdist), 0, 120)
+					HH.apply_sonic_stun(0, 0, misstep, 0, 2, ear_damage, ear_tempdeaf, stamina)
+
+		src.blast_armed = FALSE
+		update_icon()
+
+	proc/set_release_pressure(pressure as num)
+		src.release_pressure = clamp(pressure, PORTABLE_ATMOS_MIN_RELEASE_PRESSURE, PORTABLE_ATMOS_MAX_RELEASE_PRESSURE)
+		return TRUE
+
+	proc/set_arm_delay(delay as num)
+		src.blast_delay = clamp(delay, MIN_BLAST_DELAY, MAX_BLAST_DELAY)
+		return TRUE
+
+	proc/set_process_rate(rate as num)
+		src.process_rate = clamp(rate, 1, 5)
```

Comment:
Would need to validate the range is correct based on emag status.

Reply:
Updated to ternary to not trust UI.